src/Listeners/Plugin.php

```diff
@@ -44,7 +56,212 @@ public function deactivated( $plugin, $network_wide ) {
 		$data = array(
 			'plugin'       => $plugin,
 			'network_wide' => $network_wide,
+			'plugins'      => $this->collect_plugin($plugin, false),
 		);
 		$this->push( 'plugin_deactivated', $data );
 	}
+
+	/**
+	 * Plugin deleted
+	 *
+	 * @param string  $plugin Name of the plugin
+	 * @param boolean $deleted Whether the plugin deletion was successful
+	 * @return void
+	 */
+	public function deleted( $plugin, $deleted ) {
+		// abort if not successfully deleted
+		if ( !$deleted ) {
+			return;
+		}
+		
+		$slug = self::glean_plugin_slugname( $plugin );
+		$data = array(
+			'plugin' => $slug,
+		);
+		$this->push( 'plugin_deleted', $data );
+	}
+
+	/**
+	 * Plugin install or update completed
+	 *
+	 * @param string  $plugin Name of the plugin
```

Comment:
@circlecube non-blocking but the params in this docblock don't match

Reply:
Thanks, good catch here!